File: 알고리즘/전승민/numberBaseball/numberBaseball.kt
fun numberBaseball(baseball: Array<IntArray>): Int {
    var answer = 0
    (123..987).forEach { number ->
        val str = number.toString()
        if(str[0]==str[1]||str[1]==str[2]||str[0]==str[2]||str[0]=='0'||str[1]=='0'||str[2]=='0') return@forEach
        var check = true
        baseball.forEach {
            var strike = 0
            var ball = 0
            (0..2).forEach {i ->
                (0..2).forEach {j ->
                    if(str[i]==it[0].toString()[j]) {
                        if(i==j) strike++
                        else ball++
                    }
                }
            }
            if(strike!=it[1]||ball!=it[2]) check = false
        }
        if(check) answer++
    }
    return answer
}
// https://programmers.co.kr/learn/courses/30/lessons/42841

File: 알고리즘/최이삭/NumberBaseball.js
function solution(baseball) {
  let answer = 0;
  for(let i = 123; i < 987; i++) {
      let str = i.toString()
      if(str[0] === str[1] || str[0] === str[2] || str[1] === str[2]) continue;
      if(str[0] === "0" || str[1] === "0" || str[2] === "0") continue;
      let flag = true;
      baseball.every(el => {
          let target = el[0].toString();
          let strike = 0;
          let ball = 0;
          for(let j = 0; j < 3; j++) {
              for(let k = 0; k < 3; k++) {
                  if(j === k && target[j] === str[k]) strike++;
                  else if(j !== k && target[j] === str[k]) ball++;
              }
          }
          flag = el[1] === strike && el[2] === ball;
          return flag;
      })
      if(flag === true) answer++;
  }
  return answer;
}